server/server.go

```diff
@@ -590,6 +590,13 @@ func (server *ArgoCDServer) Run(ctx context.Context, listeners *Listeners) {
 	} else {
 		httpS = server.newHTTPServer(ctx, server.ListenPort, grpcWebS, appResourceTreeFn, listeners.GatewayConn, metricsServ)
 	}
+
+	if httpsS != nil {
+		httpsS.Handler = server.strictTransportSecurityMiddleware(server.hostValidationMiddleware(httpsS.Handler))
+	}
+
+	httpS.Handler = server.hostValidationMiddleware(httpS.Handler)
```

Comment:
[q] should the middleware order be reconsidered? It might be better to perform host validation before adding the HSTS header, so that rejected requests (400 errors) don't receive the Strict-Transport-Security header.